src/h5web/vis-packs/core/line/DataCurve.tsx

```diff
@@ -38,22 +38,21 @@ function DataCurve(props: Props) {
     curveType = CurveType.LineOnly,
   } = props;
 
-  const { domElement } = useThree((state) => state.gl);
-  const curveColor = color.startsWith('--')
-    ? window.getComputedStyle(domElement).getPropertyValue(color).trim()
-    : color;
-
+  const [dataGeometry] = useState(() => new BufferGeometry());
   const points = useCanvasPoints(abscissas, ordinates, errors);
 
-  const dataGeometry = useMemo(() => {
-    const geometry = new BufferGeometry();
-    geometry.setFromPoints(points.data);
-    return geometry;
-  }, [points.data]);
+  useLayoutEffect(() => {
+    dataGeometry.setFromPoints(points.data);
+  }, [dataGeometry, points.data]);
```

Comment:
Minor optim: create `BufferGeometry` once and update it with `useLayoutEffect`, similarly to what is done in `ErrorBars` (except I don't use a `ref` here because we still need to create the geometry ourselves since it's shared between two meshes).

Reply:
I thought I could improve on `setFromPoints` by going lower level (by creating the `Float32Array`s and `BufferAttribute`s myself - cf. https://github.com/mrdoob/three.js/blob/master/src/core/BufferGeometry.js#L263), but I did a lot of benchmarking and couldn't justify the change, so I reverted.